python/GafferOSL/FocalBlur.gfr

```diff
@@ -0,0 +1,889 @@
+__children["FocalBlur"]["Dot26"]["in"].setInput( __children["FocalBlur"]["Dot25"]["out"] )
+__children["FocalBlur"]["Dot26"]["__uiPosition"].setValue( imath.V2f( 822.9505, 35.2791519 ) )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["cameraPath"], 'nodule:type', '' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["cameraPath"], 'stringPlugValueWidget:placeholderText', 'Render Camera' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["cameraPath"], 'layout:section', 'Settings' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["cameraPath"], 'layout:index', 7 )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["cameraPath"], 'layout:visibilityActivator', 'cameraScene' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["cameraPath"], 'description', 'The scene path to the camera in `cameraScene` that you want to take settings from. If unset, defaults to the render camera set in the scene globals.' )
+__children["FocalBlur"]["Expression"]["__in"]["p0"].setInput( __children["FocalBlur"]["cameraPath"] )
+__children["FocalBlur"]["Expression"]["__uiPosition"].setValue( imath.V2f( 775.942566, 29.858284 ) )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["depthChannel"], 'nodule:type', '' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["depthChannel"], 'layout:section', 'Depth' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["depthChannel"], 'layout:index', 18 )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["depthChannel"], 'description', 'Which image channel to use as the depth when computing the amount of blur.' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["depthInterpretation"], 'nodule:type', '' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["depthInterpretation"], 'layout:section', 'Depth' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["depthInterpretation"], 'plugValueWidget:type', 'GafferUI.PresetsPlugValueWidget' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["depthInterpretation"], 'preset:Unpremultiplied', 'unpremultiplied' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["depthInterpretation"], 'preset:Premultiplied', 'premultiplied' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["depthInterpretation"], 'preset:Filtered With Background Value ( Arnold )', 'filteredWithBackground' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["depthInterpretation"], 'layout:index', 19 )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["depthInterpretation"], 'description', 'When using the depth channel, we need to know how it was stored? Is it premultiplied by the alpha ( typical when is set to write a depth image with a regular pixel filter )? Not premultiplied ( typical when using a pixel filter like "closest" )? Blended together with a weird massive background value set on all empty pixels ( Arnold does this when writing a filtered depth ).' )
```

Comment:
I don't mind `depthInterpretation` as a plug name, though maybe the description could be improved by approaching it roughly along these lines?

```
How the depth channel should be interpreted :

- Premultiplied : Depth has been premultiplied by the alpha (typical when writing depth with a regular pixel filter).
- Unpremultiplied : Depth has not been premultiplied by the alpha (typical when writing depth with a pixel filter such as "closest").
- Premultiplied with background value (Arnold) : Depth includes a constant value for empty pixels, which needs to be subtracted before unpremultiplying. (typical with Arnold filtered depth which includes the camera far clip plane).
```